storage/perfschema/table_malloc_stats.cc

```diff
@@ -0,0 +1,270 @@
+/* Copyright (c) 2008, 2011, Oracle and/or its affiliates. All rights reserved.
+   Copyright (c) 2016 Uber Technologies, Inc.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; version 2 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software Foundation,
+  51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA */
+
+#include <atomic>
+
+#include "storage/perfschema/table_malloc_stats.h"
+
+#include "sql/field.h"
+#include "sql/plugin_table.h"
+#include "sql/sql_profile.h"
+#include "storage/perfschema/pfs_column_types.h"
+#include "storage/perfschema/pfs_column_values.h"
+#include "storage/perfschema/pfs_global.h"
+#include "storage/perfschema/pfs_instr.h"
+
+THR_LOCK table_malloc_stats_totals::m_table_lock;
+
+static std::atomic<uint64_t> epoch{0};
+
+static void set_jemalloc_epoch() {
+  auto epoch_copy = epoch++;
```

Comment:
I don't know if it makes any difference, but for the first time epoch_copy will have the value of 0, and it will be used in jemalloc_mallctl call.
Before converting it to atomic, we used incremented value here.